### DISCUSSION TITLE
Enable Pan :- Panning Pivot Point

### DISCUSSION THREAD

**Discussion by julian-leahy** [google/model-viewer#3344] [Q&A]
I am using the viewer on a touch screen device only. Is there a way to keep the pivot point at the center of the model when using the enable pan feature. Currently if I move my model off to the side the pivot point remains at the center of the viewport so when I rotate it it rotates on a massive arc and sometime disappears out of view.

**Comment by elalish** [comment]
That's actually kind of the whole point of pan (to orbit a different point on the model), and we did several things to make it easier. For instance, tapping off of the model resets to the model center. If you don't want the pivot point to move at all, I'd recommend disabling pan.

**Reply by julian-leahy** [reply]
Thanks elalish - I appreciate what you are saying - we have problems when we are zoomed in on a model and part of it is off screen - we would like the user to be able to drag the model back into view and rotate as normal

**Reply by elalish** [reply]
Perhaps you can share a video of the problem? Either panning back or tapping the background should do what you want.

**Reply by julian-leahy** [reply]
https://user-images.githubusercontent.com/67786539/162076277-40ae3327-f534-40f2-be5f-96775c9b6b61.mov

In this example we want the user to be able to zoom in on the top of the model and be able to rotate it. As I zoom in, the top leaves the viewport - if I tap the background I get the model back into view but loose the zoom. Currently I am using the hotspot option - but I need to do it with panning (as this is the behaviour of our previous viewer) and retain the pivot point.

**Reply by elalish** [reply]
I must be missing something; it seems like the end of your video is showing your desired behavior: panning the top of the model into view. If you then rotated, you'd be rotating around the top. Can you describe the problem a little more fully?

**Comment by julian-leahy** [comment]
https://user-images.githubusercontent.com/67786539/167316504-245478f3-513e-4eb7-a32f-866fdcfd86ab.mov

This is similar to what I need to implement (taken from our current viewer) - I need to be able to pan up and down so I can view the model when it's been zoomed off the screen. Panning in the model viewer allows me to do this. The issue is when the model is panned left / right - this creates a large arc where the camera stays in the centre of the canvas and the object moved away - which as you explained is how panning works - however I need it to keep the camera in the centre of the model. I'm not really sure how to achieve this. For the log example I only need up / down - is there a way to disable left / right?

**Reply by elalish** [reply]
Okay, I think I understand now, thanks. You could either design your own pan, using pointerEvent listeners that constrains the `camera-target` to only moving in Y, or you can wait for the upcoming release (where I've fixed a `camera-change` event bug) and add a listener just to the `camera-change` event and use `getCameraTarget()` to project yourself back onto the vertical line you prefer.
